elixir-europe-training/Training-Metrics-Database
# feat: add user profile model

This PR moves the node assignment from being hard coded into being assigned to a "UserProfile" object. This will allow for the having multiple users for a single node.

It will:
- Add initial `UserProfile` model
- Hook `UserProfile` model to creation of `User`
- `User.get_node` now fetches `Node` from `UserProfile`

To test:
- Start the system
- Create a new user
- Make sure a corresponding UserProfile has been created
- Set the node for the corresponding UserProfile
- Log in as user and verify that the correct node is used
